src/pygeoprocessing/geoprocessing.py

```diff
@@ -2572,15 +2576,24 @@ def convolve_2d(
             "Expected raster path band sequences for the following arguments "
             f"but instead got: {bad_raster_path_list}")
 
+    signal_raster_info = get_raster_info(signal_path_band[0])
+    kernel_raster_info = get_raster_info(kernel_path_band[0])
+
+    for info_dict in [signal_raster_info, kernel_raster_info]:
+        if 1 in info_dict['block_size']:
+            raise ValueError(
+                f'{signal_path_band} has a row blocksize which can make this '
+                f'function run very slow, create a square blocksize using '
+                f'`warp_raster` or `align_and_resize_raster_stack` which '
+                f'creates square blocksizes by default')
```

Comment:
Since I mentioned it in my other PR I'll mention it here for consistency. Some not needed `f`-strings here, but it doesn't really matter.

Reply:
Hi Doug! I remember you raised this issue in the past and I changed it to only have the f string on the line that needed it, but I did a PR with james later and he found it confusing that on a multiline concatenated string that some had f's on them and others didn't and I tend to agree with that. So knowing that the f replacement is not used on all the strings in the concatenation i still prefer it because it's a consistent style with the entire string itself. So if you're okay to let that slide then great, and if not lets chat about it more.

Reply:
It is more aesthetically pleasing and I'm ultimately happy to let it slide. However, personally when I see a string start with an `f` I expect there to be a replacement happening in that string and when I can't find it, I get confused. To me, the f-string is misleading if there is nothing being replaced and I prefer it to be used with intent over aesthetics. But, pretty low stakes!